client/components/survey-container/components/question-step-mapping.tsx

```diff
@@ -0,0 +1,58 @@
+import { QuestionType } from '../types';
+import { QuestionSelectionComponentProps } from './question-step';
+import SurveyCheckboxControl from './survey-checkbox-control';
+import SurveyFlowCardControl from './survey-flow-card';
+import SurveyRadioControl from './survey-radio-control';
+
+type questionTypeComponentMapProps = {
+	[ QuestionType.SINGLE_CHOICE ]: ( {
+		onChange,
+		question,
+		value,
+		disabled,
+		onContinue,
+	}: QuestionSelectionComponentProps ) => JSX.Element;
+	[ QuestionType.MULTIPLE_CHOICE ]: ( {
+		onChange,
+		question,
+		value,
+		disabled,
+		onContinue,
+	}: QuestionSelectionComponentProps ) => JSX.Element;
+};
+
+type questionHeaderButtonsMapProps = {
+	[ key: string ]: {
+		shouldHideContinueButton: boolean;
+		shouldHideSkipButton: boolean;
+	};
+};
+
+export const questionTypeComponentMap = {
+	[ QuestionType.SINGLE_CHOICE ]: SurveyRadioControl,
+	[ QuestionType.MULTIPLE_CHOICE ]: SurveyCheckboxControl,
+};
+
+export const flowQuestionMapping: {
+	[ key: string ]: {
+		questionTypeComponentMap: questionTypeComponentMapProps;
+		headerButtonsMapping: questionHeaderButtonsMapProps;
+	};
+} = {
+	'/start/guided/initial-intent': {
```

Comment:
It would be nice if we could make it non-dependant on the path; it could scale better in case other flows decide this set of components in the future.